com.unity.render-pipelines.core/Runtime/Debugging/DebugUI.Fields.cs

```diff
@@ -279,15 +283,23 @@ public Type autoEnum
             {
                 set
                 {
-                    enumNames = Enum.GetNames(value).Select(x => new GUIContent(x)).ToArray();
-
-                    // Linq.Cast<T> on a typeless Array breaks the JIT on PS4/Mono so we have to do it manually
-                    //enumValues = Enum.GetValues(value).Cast<int>().ToArray();
-
                     var values = Enum.GetValues(value);
                     enumValues = new int[values.Length];
+                    enumNames = new GUIContent[values.Length];
                     for (int i = 0; i < values.Length; i++)
-                        enumValues[i] = (int)values.GetValue(i);
+                    {
+                        var enumValue = values.GetValue(i);
+                        var memInfo = value.GetMember(value.GetEnumName(enumValue));
+                        var name = memInfo[0]
+                            .GetCustomAttributes(typeof(InspectorNameAttribute), false)
+                            .FirstOrDefault() is InspectorNameAttribute attribute ? attribute.displayName : enumValue.ToString();
+
+                        if (memInfo[0].GetCustomAttributes(typeof(ObsoleteAttribute), false).FirstOrDefault() is ObsoleteAttribute)
+                            name += "(Obsolete)";
```

Comment:
missing a space here? "blabla that do thing(obsolete)" -> "blabla that do thing (obsolete)"

Reply:
changed